### ISSUE TITLE
Pause key not working in PCjr game Demon Attack

### ISSUE THREAD

**Issue by MX9000** [joncampbell123/dosbox-x#4794]
### Describe the bug

The manual says that you can interrupt the game with the PAUSE key on the PCjr keyboard.
eXoDOS uses DOSBox Staging version 0.81.0-alpha-1770-g904cb68f9: the game pauses with the Fn+Pause/Break key combination and resumes by pressing the Num Lock key.
In DOSBox-X the game freezes often with a hanging note and no longer comes out of pause.
I also tried using Ctrl+Num Lock as recommended here: http://brutmanlabs.org/PCjr/pcjr_keyboard_diff.html.

### Steps to reproduce the behaviour

1. Boot Demon Attack cartridge
2. Try to pause and unpause the game in DOSBox-X and eXoDOS

### Expected behavior

The game should go in and out of pause.

### What operating system(s) this bug have occurred on?

Windows 10 Pro 22H2 64 bit

### What version(s) of DOSBox-X have this bug?

DOSBox-X 2023.10.06 MinGW & VSbuild

### Used configuration

```ini
machine = pcjr
cycles = 300
```

### Output log

```text
No errors shown
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
It looks like the PCjr never had a Pause key anyway. https://deskthority.net/wiki/images/f/f8/Pcjr_front.jpg so the key shouldn't do anything.

However if non-PCjr modes have a problem with Pause,  that's an issue.

**Comment by MX9000** [comment]
Yes, @Torinde, also the first version of the PCjr keyboard has a Pause function associated with key combination Fn+Q.
Thanks.

![keyboard_chiclet_large](https://github.com/joncampbell123/dosbox-x/assets/58113425/39f3ede9-0e86-48df-99f7-c80f51382dc1)
